ccf_transaction_processor/README.md

```diff
@@ -0,0 +1,208 @@
+located at `$PDO_HOME/ccf/keys`. The network certificate is
+`networkcert.pem`. User public certificate is `user0_cert.pem` and
+private key is `user0_privk.pem`.  Note that the keys are created as
+part of CCF deployment and are unique to the specific instance of CCF.
+
+In our usage, CCF users are PDO clients, and PDO client authentication
+is implemented within the transaction processor itself. Thus, we do not
+utilize the client authentication feature provided by CCF. However to
+satisfy the CCF's requirement that only authorized CCF users can submit
+transactions to CCF, share `user0_cert.pem` and `user0_privk.pem` with
+all the PDO clients. These two keys and the network certificate
+`networkcert.pem` must be stored under $PDO_LEDGER_KEY_ROOT (as part of
+PDO deployment). The user keys must be renamed as `userccf_cert.pem` and
+`userccf_privk.pem` respectively.
+
+## Test the Deployment with Ping Test
+
+PDO TP contains a simple `ping` rpc that returns success every time it
+is queried. Test the PDO-TP deployment using this rpc. Invoke the
+following commands to issue 100 ping rpcs. The net througput is reported
+at the end of the test.
+
+```bash
+${PDO_SOURCE_ROOT}/ccf_transaction_processor/scripts/ping_test.py
+```
+
+## Generate Ledger Authority Key
+
+Responses to read-transactions include a payload signature, where the
+signature is generated within PDO-TP.  The required signing keys must be
+generated before PDO-TP can be opened up for business from PDO
+clients. This will be done automatically if you start the CCF network
+with the script `${PDO_HOME}/ccf/bin/start_ccf_network.sh`.
+
+Otherwise, you may invoke the following commands to generate and save
+the ledger authority key.
+
+```bash
+${PDO_HOME}/ccf/bin/generate_ledger_authority.py
+${PDO_HOME}/ccf/bin/fetch_ledger_authority.py
+```
+
+If successful, the rpc returns after 1) creating one set of signing keys
+locally within the CCF enclave, and 2) scheduling them for global
+commit. The ledger authority verifying key can be obtained using the
+`get_ledger_verifying_key` rpc. The verifying key is returned only after
+global commit is successful.
+
+The read-payload-signature feature may be used by PDO clients to
+establish offline verifiable proof of transaction commits as desired by
+the PDO smart contract application. Note that for trust purposes, it is
+recommended that any entity that uses the verifying_key gets it directly
+from the CCF service using the `get_ledger_verifying_key` rpc.
+
+## Using CCF ledger using PDO
+
+We highlight some quick details about how PDO clients can use a CCF
+based PDO-TP deployment. The information below can be found at
+[PDO docs](../docs) as well.
+
+1. Set the following environment variables:
+
+```bash
+export PDO_LEDGER_TYPE=ccf
+export PDO_LEDGER_URL=http://ip-address:6006
+```
+
+Here, ip-address is the address of the node which hosts the CCF instance.
+
+2. Set env PDO_LEDGER_KEY_ROOT and save CCF's network certificate
```

Comment:
what does PDO_LEDGER_KEY_ROOT have to be set to?!  This whole paragraph is a bit rough reading. You might also refernce back to the "Share CCF (TLS) Authentication Keys" section for more context?

Reply:
Thank you for the question. PDO_LEDGER_KEY_ROOT is the path to the folder where the keys must be saved. This paragraph is a brief of what was already explained earlier in the same README.md. Sorry for any lack of clarity, I have clarified better now.